posthog/hogql_queries/query_runner.py

```diff
@@ -349,53 +373,89 @@ def _limit_context_aliased_for_cache(self) -> LimitContext:
     def calculate(self) -> R:
         raise NotImplementedError()
 
+    def enqueue_async_calculation(
+        self, *, cache_key: str, refresh_requested: bool = False, user: Optional[User] = None
+    ) -> QueryStatusResponse:
+        query_status = enqueue_process_query_task(
+            team=self.team,
+            user=user,
+            query_json=self.query.model_dump(),
+            query_id=self.query_id or cache_key,  # Use cache key as query ID to avoid duplicates
+            refresh_requested=refresh_requested,
+        )
+        return QueryStatusResponse(query_status=query_status)
+
+    def retrieve_from_cache(
```

Comment:
The name is a bit of a misnomer, because it might either be retrieval from cache, or kicking off async execution

Reply:
Working on the hardest problems here 😅 Caching and naming.
Will rethink.